api/routes_payment.go

```diff
@@ -1,69 +1,317 @@
 package api
 
 import (
+	"context"
+	"encoding/hex"
 	"errors"
+	"fmt"
 	"net/http"
+	"strconv"
 
 	"github.com/RTradeLtd/Temporal/eh"
+	"github.com/jinzhu/gorm"
+
+	"github.com/RTradeLtd/Temporal/gapi"
+
 	"github.com/RTradeLtd/Temporal/models"
 	"github.com/RTradeLtd/Temporal/queue"
 	"github.com/RTradeLtd/Temporal/utils"
+	greq "github.com/RTradeLtd/gapimit/request"
 	"github.com/gin-gonic/gin"
 )
 
-// CreatePayment is used to create a payment
-func (api *API) CreatePayment(c *gin.Context) {
+// ConfirmPayment is used to confirm a payment after sending it.
+// By giving Temporal the TxHash, we can then validate that hte payment
+// was made, and validated by the appropriate blockchain.
+func (api *API) ConfirmPayment(c *gin.Context) {
+	username := GetAuthenticatedUserFromContext(c)
+	paymentNumber, exists := c.GetPostForm("payment_number")
+	if !exists {
+		FailWithMissingField(c, "payment_number")
+		return
+	}
+	txHash, exists := c.GetPostForm("tx_hash")
+	if !exists {
+		FailWithMissingField(c, "tx_hash")
+		return
+	}
+	paymentNumberInt, err := strconv.ParseInt(paymentNumber, 10, 64)
+	if err != nil {
+		Fail(c, err)
+		return
+	}
+	pm := models.NewPaymentManager(api.dbm.DB)
```

Comment:
can we initialize `pm` with `api` and hold on to it as a property of `API`?

Reply:
Yup! I'll just transition to having all "model managers" as properties of `API` ?

Reply:
sounds good!